src/input/isf.c

```diff
@@ -0,0 +1,796 @@
+{
+	char byte_order_buf[BYTE_ORDER_BUFFER_SIZE];
+	char format_buf[DATA_FORMAT_BUFFER_SIZE];
+	int ret;
+
+	switch (item) {
+	case YOFF:
+		if (!str_to_float(buf, buflen, &inc->yoff))
+			return SR_ERR_DATA;
+		break;
+
+	case YZERO:
+		if (!str_to_float(buf, buflen, &inc->yzero))
+			return SR_ERR_DATA;
+		break;
+
+	case YMULT:
+		if (!str_to_float(buf, buflen, &inc->ymult))
+			return SR_ERR_DATA;
+		break;
+
+	case XINCR:
+		if (!str_to_float(buf, buflen, &inc->xincr))
+			return SR_ERR_DATA;
+		break;
+
+	case BYTNR:
+		if (!str_to_uint(buf, buflen, &inc->bytnr))
+			return SR_ERR_DATA;
+		break;
+
+	case BYTE_ORDER:
+		find_string_value(buf, buflen, byte_order_buf, BYTE_ORDER_BUFFER_SIZE);
+		if (strcmp(byte_order_buf, "LSB") == 0)
+			inc->byte_order = LSB;
+		else if (strcmp(byte_order_buf, "MSB") == 0)
+			inc->byte_order = MSB;
+		else
+			return SR_ERR_DATA;
+		break;
+
+	case BN_FMT:
+		find_string_value(buf, buflen, format_buf, DATA_FORMAT_BUFFER_SIZE);
+		if (strcmp(format_buf, "RI") == 0)
+			inc->bn_fmt = RI;
+		else if (strcmp(format_buf, "RP") == 0)
+			inc->bn_fmt = RP;
+		else if (strcmp(format_buf, "FP") == 0)
+			inc->bn_fmt = FP;
+		else
+			return SR_ERR_DATA;
+		break;
+
+	case WFID:
+		extract_channel_name(inc, buf, buflen);
+		break;
+
+	case WFMTYPE:
+		ret = find_waveform_type(inc, buf, buflen);
+		if (ret != SR_OK)
+			return ret;
+		break;
+
+	case ENCODING:
+		ret = find_encoding(buf, buflen);
+		if (ret != SR_OK)
+			return ret;
+		break;
+	default:
+		return SR_ERR_ARG;
+	}
+	
+	return SR_OK;
+}
+
+/* Parse the input file header. */
+static int parse_isf_header(GString *buf, struct context *inc)
+{
+	char *pattern, *data_section;
+	int ret, i;
+	size_t item_offset, data_section_offset;
+
+	if (inc == NULL)
+		return SR_ERR_ARG;
+
+	data_section = find_data_section(buf);
+	if (data_section == NULL)
+		return SR_ERR_DATA;
+	data_section_offset = (size_t) (data_section - buf->str);
+
+	/* Search for all header items. */
+	for (i = 0; i < HEADER_ITEMS_PARAMETERS; ++i) {
+		pattern = find_item(buf->str, data_section_offset, header_items[i]);
+		if (pattern == NULL) {
+			/* WFID is not required. */
+			if (i == WFID)
+				continue;
+			return SR_ERR_DATA;
+		}
+
+		/*
+		 * Calculate the offset of the header item value in the buffer
+		 * as well as its distance from the data section.
+		 */
+		item_offset = (size_t) (pattern - buf->str);
+		item_offset += strlen(header_items[i]);
+		if (item_offset >= data_section_offset)
+			return SR_ERR_DATA;
+
+		ret = process_header_item(buf->str + item_offset, data_section_offset - item_offset, inc, i);
+		if (ret != SR_OK)
+			return ret;
+	}
+
+	return SR_OK;
+}
+
+/*
+ * Check if the format matches ISF format.
+ *
+ * TODO: The header could be searched for more items
+ * aside from NR_PT to increase the confidence.
+ */
+static int format_match(GHashTable *metadata, unsigned int *confidence)
+{
+	const char default_extension[] = ".isf";
```

Comment:
suggest to do extension matching with non-case-sensitive functions (some files may end up with uppercase .ISF and then fail on linux/mac)

Reply:
The extension matching is now done using a non-case-sensitive function.